src/test/kotlin/graphics/scenery/tests/examples/compute/PersistentTextureRequestsExample.kt

```diff
@@ -0,0 +1,143 @@
+import tpietzsch.shadergen.generate.SegmentTemplate
+import tpietzsch.shadergen.generate.SegmentType
+import java.util.concurrent.atomic.AtomicInteger
+import kotlin.concurrent.thread
+import kotlin.test.assertEquals
+
+/**
+ * Example to show how persistent texture requests - that are served once per frame - may be created
+ * and accessed. These may be particularly be useful, e.g, when compute shaders are used and output
+ * textures need to be accessed every time they are updated.
+ *
+ * @author Aryaman Gupta <argupta@mpi-cbg.de>
+ * @author Ulrik Günther <hello@ulrik.is>
+ */
+class PersistentTextureRequestsExample : SceneryBase("PersistentTextureRequestsExample") {
+
+    private val counter = AtomicInteger(0)
+    private var totalFrames = -1L
+
+    override fun init() {
+        renderer = hub.add(
+            SceneryElement.Renderer,
+            Renderer.createRenderer(hub, applicationName, scene, 1280, 720))
+
+        val volumeManager = VolumeManager(hub,
+            useCompute = true,
+            customSegments = hashMapOf(
+                SegmentType.FragmentShader to SegmentTemplate(
+                    this.javaClass,
+                    "ComputeVolume.comp",
+                    "intersectBoundingBox", "vis", "SampleVolume", "Convert", "Accumulate"),
+            ))
+
+        val outputBuffer = MemoryUtil.memCalloc(1280*720*4)
+        val outputTexture = Texture.fromImage(Image(outputBuffer, 1280, 720), usage = hashSetOf(Texture.UsageType.LoadStoreImage, Texture.UsageType.Texture))
+        volumeManager.material.textures["OutputRender"] = outputTexture
+
+        hub.add(volumeManager)
+
+        val imp: ImagePlus = IJ.openImage("https://imagej.nih.gov/ij/images/t1-head.zip")
+        val img: Img<UnsignedShortType> = ImageJFunctions.wrapShort(imp)
+
+        val volume = Volume.fromRAI(img, UnsignedShortType(), AxisOrder.DEFAULT, "T1 head", hub, VolumeViewerOptions())
+        volume.transferFunction = TransferFunction.ramp(0.001f, 0.5f, 0.3f)
+        scene.addChild(volume)
+
+        val box = Box(Vector3f(1.0f, 1.0f, 1.0f))
+        box.name = "le box du win"
+        box.material.textures["diffuse"] = outputTexture
+        box.material.metallic = 0.0f
+        box.material.roughness = 1.0f
+
+        scene.addChild(box)
+
+        val light = PointLight(radius = 15.0f)
+        light.position = Vector3f(0.0f, 0.0f, 2.0f)
+        light.intensity = 5.0f
+        light.emissionColor = Vector3f(1.0f, 1.0f, 1.0f)
+        scene.addChild(light)
+
+        val cam: Camera = DetachedHeadCamera()
+        with(cam) {
+            position = Vector3f(0.0f, 0.0f, 5.0f)
+            perspectiveCamera(50.0f, 512, 512)
+
+            scene.addChild(this)
+        }
+
+        thread {
+            val opTexture = volumeManager.material.textures["OutputRender"]!!
+
+            var prevCounter = 0
+
+            (renderer as? VulkanRenderer)?.persistentTextureRequests?.add(opTexture to counter)
```

Comment:
@skalarproduktraum as we have discussed, the `PersistentTextureRequests` are implemented only in VulkanRenderer, as the OpenGLRenderer does not support compute shaders anyway. Should this example therefore be put in the `blocklist` of the ExampleRunner?

Reply:
Not necessary to add it to the blocklist, because the CI only runs the VulkanRenderer anyway.